### PR TITLE
Extending Alerters Docs

### PR DESCRIPTION
**Pull request by fa9r**
## Describe changes
I added documentation on extending alerters. In the process, I also improved some of the alerter docstrings.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/features/integrations) table.
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Other (add details above)

### CODE REVIEW

File: docs/book/extending-zenml/alerter.md [hunk 1]
```diff
@@ -2,31 +2,79 @@
...
+concept of [stacks, stack components and their flavors](../advanced-guide/stacks-components-flavors.md).  
+{% endhint %}
+
+## Base Abstraction
+
+The base abstraction for alerters is very simple, as it only defines two
+abstract methods that subclasses should implement:
+- `post()` takes a string, posts it to the desired ChatOps tool, and returns 
+`True` if the operation succeeded, else `False`.
+- `ask()` does the same as post, but after sending the message, it waits
```
Comment (strickvl) [inline docs/book/extending-zenml/alerter.md:20]:
```suggestion
- `ask()` does the same as `post`, but after sending the message, it waits
```

File: docs/book/extending-zenml/alerter.md [hunk 2]
```diff
@@ -2,31 +2,79 @@
...
-When it comes to alerters, ZenML comes with an integration into Slack, which 
-can be used to post messages to a specified channel.
+Currently, the only available alerter integration is Slack.
+The Slack alerter can be used to post/ask in a specified Slack channel via a 
+dedicated bot account of a Slack app linked to the channel.
 
 |                | Flavor | Integration |
 |----------------|--------|-------------|
-| SlackAlerter   | slack  | slack       |
+| [SlackAlerter](https://apidocs.zenml.io/0.8.1/api_docs/integrations/#zenml.integrations.slack.alerters.slack_alerter.SlackAlerter)   | slack  | slack       |
```
[the remark this answers is not in the document]
Comment (fa9r) [reply docs/book/extending-zenml/alerter.md:58]: Good catch, changed it to latest.

File: docs/book/extending-zenml/alerter.md (hunk 1 above)
Comment (fa9r) [reply docs/book/extending-zenml/alerter.md:20]: changed to `post()`

File: docs/book/extending-zenml/alerter.md [hunk 3]
```diff
@@ -2,31 +2,79 @@
 description: Sending alerts through specified channels.
 ---
 
-Alerters allow you to send alerts from within your pipeline. This is especially 
-useful to immediately get notified when failures happen, and also for 
-general monitoring/reporting.
+Alerters allow you to send messages to ChatOps tools from within your pipeline.
```
[the remark this answers is not in the document]
Comment (fa9r) [reply docs/book/extending-zenml/alerter.md:5]: Yeah, I don't like the term either, there also seems to be no clear definition for it. I renamed it to "chat services" everywhere.

File: docs/book/advanced-guide/alerts.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
+---
+description: Send automated alerts to chat services.
+---
+
+# :speech_balloon: Send Automated Chat Alerts
+
+Many developer teams use real-time chat services like Slack for their
+day-to-day communication. Sometimes it can also be desired to have automated
+notifications be sent to those chat services from within ML pipelines,
```
Comment (strickvl) [inline docs/book/advanced-guide/alerts.md:9]:
```suggestion
day-to-day communication. Members of those teams might also want to have automated
notifications be sent to those chat services from within ML pipelines,
```

**Review by strickvl — APPROVED** [review]
I made one extra phrasing suggestion, but otherwise this looks good to me!

**Review by htahir1 — APPROVED** [review]
Awesome!

**Pull request merged**